java-manta-it/src/test/java/com/joyent/manta/client/helper/IntegrationTestHelper.java

```diff
@@ -73,10 +96,24 @@ public static void createTestBucketOrDirectory(final MantaClient client,
                                       final String testPath,
                                       final String testType) throws IOException {
         if ("buckets".equals(testType)) {
+            final String bucketPath = parseBucketPath(testPath);
             //Create bucket here
+            try {
+                client.createBucket(bucketPath);
+            } catch (MantaClientHttpResponseException e) {
+                if (e.getServerCode().equals(MantaErrorCode.BUCKET_EXISTS_ERROR)) {
+                    LOG.error("Test Bucket named {} already exists leading to {}"
+                            , integrationTestBucket, e.getStatusMessage());
+                    e.setContextValue("Creating test bucket failed at :", testPath);
+                    throw e;
+                }
```

Comment:
Why is the exception swallowed here and not thrown? If there is ANY problem creating a bucket, the exception for that problem is lost at this point.

Reply:
Yes noted feedback taken.